src/main.ts

```diff
@@ -37,16 +39,49 @@ export class Minifier {
 
   // Recursively visits every child node, emitting text of the sourcefile that is not a part of
   // a child node.
-  visit(node: ts.Node) {
+  visit(node: ts.Node, typeChecker?: ts.TypeChecker) {
     switch (node.kind) {
-      case ts.SyntaxKind.PropertyAccessExpression: {
-        let output = '';
-        let propAccessExp = <ts.PropertyAccessExpression>node;
-        output += this.visit(propAccessExp.expression);
-        output += '.';
-        // Adds '$mangled' to an identifier to ensure that the identifier is being altered.
-        output += propAccessExp.name.text + '$mangled';
-        return output;
+      case ts.SyntaxKind.Identifier: {
+        let parent = node.parent;
+
+        if (parent.kind === ts.SyntaxKind.PropertyDeclaration) {
+          return this.renameIdent(node);
+        } else if (parent.kind === ts.SyntaxKind.PropertyAccessExpression) {
+          let pae = <ts.PropertyAccessExpression>parent;
+          let exprSymbol = typeChecker.getTypeAtLocation(pae.expression).symbol;
+          if (!exprSymbol) {
+            exprSymbol = typeChecker.getSymbolAtLocation(pae.expression);
+          }
+          if (!exprSymbol) {
+            exprSymbol = typeChecker.getSymbolAtLocation(pae);
+          }
+          let childText = '';
+          if (exprSymbol) {
```

Comment:
You can make your code easier to read and less indented by doing early exits for error cases.

E.g. here say `if (!exprSymbol) return this.renameIdent(node);`, and then continue with the code underneath.